improv/store.py

```
@@ -114,8 +115,9 @@ def put(self, object):
             # TODO key; not sure it's worth the network overhead to check every
             # TODO key twice every time. we still need a better solution for
             # TODO this, but it will work now singlethreaded most of the time.
-
-            self.client.set(object_key, pickle.dumps(object, protocol=5), nx=True)
+            self.client.set(
+                object_key, pickle.dumps(object, protocol=5), nx=True, ex=ex
```

What is `nx` here?

---

`nx = True` means the key will only be set if it does not already exist in the redis store. If the key already exists, `set()` will be skipped.

---

Ah we should probably decide if we want default behavior to be non-overwriting or overwriting. The keys should never conflict but in the very very unlikely case they do. I think perhaps overwriting is more real-time... @jmxpearson ?

---

On the one hand, overwriting by default is begging for silent failure. On the other, speed is necessary. 

In all, I’d probably prefer not overwriting as the default but exposing the option and documenting u der some section detailing optimizations (TTL, store size, etc.)?

---

@paulawucu Let's keep nx=True then, but have it log if set is skipped? Is that possible with Redis code? Does it return any warnings? 
Could also regen a key if set is going to be skipped.

Let's also file an issue to add any error handling for if there is key collision and the user is choosing nx=False.